File: Python/BinarySearchTree.py
class Node:
    def __init__(self, data, left_node=None, right_node=None):
        self.data = data
        self.left = left_node
        self.right = right_node
    
    def set_link(self, node):
        self.link = node

    def get_link(self):
        return self.link

class BinarySearchTree:
    def __init__(self):
        self.root = None

    def insert(self, data):
        if self.root == None:
            root = Node(data)
        else:
            return insert_node(data, root)

    def insert_node(self, data, node):
        if node == None:
            node = Node(data)
        elif data < node.data:
            return insert_node(data, node.left)
        else:
            return insert_node(data, node.right)

    def delete(self, data):
        pass

    def delete_node(self, data):
        pass

    def search(self, data):
        if self.root == None:
            print("Binary Search Tree Empty !")
            return False
        else:
            return search_node(data, root)
  
    def search_node(self, data, node):
        if data == node.data:
            return True
        elif data < node.data:
            return search_node(data, node.left)
        else:
            return search_node(data, node.right)

    def size(self):
        if self.root == None:
            return 0
        else:
            return size_node(self.root.left) + size_node(self.root.right) + 1
    
    def size_node(self, node):
        if node == Node:
            return 0
        else:
            return size_node(self.root.left) + size_node(self.root.right) + 1


def main():
    binary_search_tree = BinarySearchTree()


if __name__ == "__main__":
    main()

File: Javascript/BinarySearchTree.js
class Node {
    constructor(data, leftNode, rightNode){
        this.data = data;
        this.left = leftNode;
        this.right = rightNode;
    }

    setLink = (node) => {
        this.link = node;
    }

    getLink = () => {
        return this.link;
    }
}

class BinarySearchTree {
    constructor(){
        this.root = null;
    }

    insert = (data) => {
        if (this.root === null){
            this.root = new Node(data, null, null);
        } else {
            insertNode(data, node)
        }
    }

    insertNode = (data, node) => {
        if (node === null){
            node = new Node(data, null);
        } else if (data < node.data){
            return this.insertNode(data, node.left);
        } else {
            return this.insertNode(data, node.right);
        }
    }

    // delete = () => {}
    // deleteNode = (data) => {}

    find = (data) => {
        if (this.root == null){
            console.log("Binary Search Tree Empty!");
            return false;
        } else {
            return this.findNode(data, this.root);
        }
    }

    findNode = (data, node) => {
        if (data === node.data){
            return true;
        } else if (data < node.data){
            return this.findNode(data, node.left);
        } else {
            return this.findNode(data, node.right);
        }
    }

    size = () => {
        if (this.root == null){
            return 0;
        } else {
            return this.sizeNode(this.root.left) + this.size(this.root.right) + 1;
        }
    }

    sizeNode = (node) => {
        if (node === null){
            return 0;
        } else {
            return this.sizeNode(this.root.left) + this.size(this.root.right) + 1
        }
    }    
}

const binarysearchtree = new BinarySearchTree();
